src/core/ComboBox/ComboBox.test.tsx

```diff
@@ -59,12 +64,185 @@ it('should render with selected value', () => {
+  fireEvent.change(input, { target: { value: 'Item-2' } });
+  expect(menu.children).toHaveLength(1);
+  expect(menu.firstElementChild).toHaveTextContent('Item-2');
+
+  // no items
+  fireEvent.change(input, { target: { value: 'Item-5' } });
+  expect(menu.children).toHaveLength(1);
+  expect(menu.firstElementChild).toHaveTextContent(emptyText);
+
+  // 1 item
+  fireEvent.change(input, { target: { value: 'Item-3' } });
+  expect(menu.children).toHaveLength(1);
+  expect(menu.firstElementChild).toHaveTextContent('Item-3');
+
+  // clear filter
+  fireEvent.change(input, { target: { value: '' } });
+  expect(menu.children).toHaveLength(4);
+});
+
+it('should select value on click', () => {
+  const mockOnChange = jest.fn();
+  const { container, getByText } = renderComponent({ onChange: mockOnChange });
+  const input = assertBaseElement(container);
+
+  input.focus();
+  getByText('Item 1').click();
+  expect(mockOnChange).toHaveBeenCalledWith(1);
+  expect(container.querySelector('.iui-menu')).not.toBeVisible();
+  expect(input.value).toEqual('Item 1');
+
+  input.blur();
+  input.focus();
+  expect(
+    container.querySelector('.iui-menu-item.iui-active.iui-focused'),
+  ).toHaveTextContent('Item 1');
+});
+
+it('should handle keyboard navigation', () => {
+  const id = 'test-component';
+  const mockOnChange = jest.fn();
+  const { container } = renderComponent({ id, onChange: mockOnChange });
+
+  const input = assertBaseElement(container);
+  input.focus();
+  expect(input).toHaveAttribute('aria-controls', `${id}-list`);
+
+  const items = container.querySelectorAll('.iui-menu-item');
+
+  // focus index 0
+  fireEvent.keyDown(input, { key: 'ArrowDown' });
+  expect(input).toHaveAttribute('aria-activedescendant', `${id}-option0`);
+  expect(items[0]).toHaveClass('iui-focused');
+
+  // 0 -> 1
+  fireEvent.keyDown(input, { key: 'ArrowDown' });
+  expect(input).toHaveAttribute('aria-activedescendant', `${id}-option1`);
+  expect(items[1]).toHaveClass('iui-focused');
+  expect(items[0]).not.toHaveClass('iui-focused');
+
+  // 1 -> 2
+  fireEvent.keyDown(input, { key: 'ArrowDown' });
+  expect(input).toHaveAttribute('aria-activedescendant', `${id}-option2`);
+  expect(items[2]).toHaveClass('iui-focused');
+
+  // 2 -> 2
+  fireEvent.keyDown(input, { key: 'ArrowDown' });
+  expect(input).toHaveAttribute('aria-activedescendant', `${id}-option2`);
+  expect(items[2]).toHaveClass('iui-focused');
+
+  // 2 -> 1
+  fireEvent.keyDown(input, { key: 'ArrowUp' });
+  expect(input).toHaveAttribute('aria-activedescendant', `${id}-option1`);
+  expect(items[1]).toHaveClass('iui-focused');
+  expect(items[2]).not.toHaveClass('iui-focused');
+
+  // 1 -> 0
+  fireEvent.keyDown(input, { key: 'ArrowUp' });
+  expect(input).toHaveAttribute('aria-activedescendant', `${id}-option0`);
+  expect(items[0]).toHaveClass('iui-focused');
+
+  // select 0
+  fireEvent.keyDown(input, { key: 'Enter' });
+  // expect(mockOnChange).toHaveBeenCalledWith(0); // FIXME
```

Comment:
?

Reply:
For some reason, Enter key is closing the menu but not calling `onChange` here

Reply:
Fixed